ql/src/java/org/apache/hadoop/hive/ql/DriverContext.java

```diff
@@ -125,6 +130,17 @@ public QueryPlan getPlan() {
 
   public void setPlan(QueryPlan plan) {
     this.plan = plan;
+    // only set runtimeContext if the plan is not null
+    // we don't want to nullify runtimeContext if this method is called with plan=null, which is the case when e.g.
+    // driver.releasePlan() tries to release resources/objects that are known to be heavy
+    if (plan != null) {
+      TezTask task = Utilities.getFirstTezTask(plan.getRootTasks()).orElse(null);
+      this.runtimeContext = task == null ? null : task.getRuntimeContext();
```

Comment:
minor style: i usually wrap ternary condition in round brackets, unless it's single boolean var 
````
this.runtimeContext = (task == null) ? null : task.getRuntimeContext();
````